### PR TITLE
Small cleanup in SD::BatchOptimizer

### PR DESCRIPTION
**Pull request by jppelteret**

### CODE REVIEW

File: include/deal.II/differentiation/sd/symengine_optimizer.h [hunk 1]
```diff
@@ -1477,16 +1477,15 @@ namespace Differentiation
...
+       * Copy constructor.
        *
-       * The @p copy_initialized flag, which is set to <code>true</code> by default,
-       * determines whether or not all of the optimized data is copied over from
-       * the @p other optimizer instance. Only with the flag set to <code>false</code>
-       * is it possible to re-optimize the data stored in this class with a
-       * different optimization scheme.
-       */
-      BatchOptimizer(const BatchOptimizer &other/*,
-                     const bool            copy_initialized = true*/);
```
Comment (jppelteret) [inline include/deal.II/differentiation/sd/symengine_optimizer.h:1489]: Note: No implementation change here: This just removes a commented out parameter that was a legacy of the prototype implementation.

Comment (bangerth) [reply include/deal.II/differentiation/sd/symengine_optimizer.h:1489]: Oh, I had not even realized that the parameter was in reality non-existent :-)

**Review by bangerth — APPROVED** [review]

File: include/deal.II/differentiation/sd/symengine_optimizer.h (hunk 1 above)
Comment (jppelteret) [reply include/deal.II/differentiation/sd/symengine_optimizer.h:1489]: No worries :-) It was an unusual change to review. I shouldn't have been so sloppy in the first place.

**Pull request merged**